plugins/crd/plugin_impl_crd.go

```diff
@@ -0,0 +1,345 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// //go:generate protoc -I ./model/pod --go_out=plugins=grpc:./model/pod ./model/pod/pod.proto
+
+package netmeshplugincrd
+
+import (
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	apiextv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	apiextcs "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/workqueue"
+
+	"github.com/ligato/cn-infra/config"
+	"github.com/ligato/cn-infra/flavors/local"
+	"github.com/ligato/cn-infra/health/statuscheck"
+	"github.com/ligato/cn-infra/logging"
+	"github.com/ligato/networkservicemesh/pkg/apis/networkservicemesh.io/v1"
+	client "github.com/ligato/networkservicemesh/pkg/client/clientset/versioned"
+	factory "github.com/ligato/networkservicemesh/pkg/client/informers/externalversions"
+)
+
+// Plugin watches K8s resources and causes all changes to be reflected in the ETCD
+// data store.
+type Plugin struct {
+	Deps
+
+	stopCh chan struct{}
+	wg     sync.WaitGroup
+
+	k8sClientConfig *rest.Config
+	k8sClientset    *kubernetes.Clientset
+	apiclientset    *apiextcs.Clientset
+	crdClient       client.Interface
+
+	StatusMonitor statuscheck.StatusReader
+}
+
+// Deps defines dependencies of netmesh plugin.
+type Deps struct {
+	local.PluginInfraDeps
+	// Kubeconfig with k8s cluster address and access credentials to use.
+	KubeConfig config.PluginConfig
+}
+
+var (
+	// queue is a queue of resources to be processed. It performs exponential
+	// backoff rate limiting, with a minimum retry period of 5 seconds and a
+	// maximum of 1 minute.
+	queueNS  = workqueue.NewRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(time.Second*5, time.Minute))
+	queueNSC = workqueue.NewRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(time.Second*5, time.Minute))
+	queueNSE = workqueue.NewRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(time.Second*5, time.Minute))
+
+	// stopCh can be used to stop all the informer, as well as control loops
+	// within the application.
+	stopCh = make(chan struct{})
+
+	// sharedFactory is a shared informer factory that is used a a cache for
+	// items in the API server. It saves each informer listing and watching the
+	// same resources independently of each other, thus providing more up to
+	// date results with less 'effort'
+	sharedFactoryNS  factory.SharedInformerFactory
+	sharedFactoryNSE factory.SharedInformerFactory
+	sharedFactoryNSC factory.SharedInformerFactory
+)
+
+// Init builds K8s client-set based on the supplied kubeconfig and initializes
+// all reflectors.
+func (plugin *Plugin) Init() error {
```

Comment:
Note that having to run the Init() routine (since this is a Ligato plugin) may not be obvious to Ligato plugin authors. May be worth looking into this in cn-infra.